core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -765,7 +824,7 @@ public ReplicationPhase(ReplicaRequest replicaRequest, Response finalResponse, S
                     if (shard.currentNodeId().equals(nodes.localNodeId()) == false) {
                         numberOfPendingShardInstances++;
                     }
-                    if (shard.relocating()) {
+                    if (shard.relocating() && shard.relocatingNodeId().equals(nodes.localNodeId()) == false) {
```

Comment:
done.

alternatively, the list of shards to replicate to could be built in the constructor.
We would then only iterate over the list here.